<?xml version="1.0"?>
<launch>

  <!-- start camera -->
  <include file="$(find smocap)/launch/ueye_cam.launch"/>
  
  <!-- spawn marker localization node -->
  <node name="smocap_node" pkg="smocap" type="node">
    <param name="show_gui"        type="bool"   value="false" />
    <param name="publish_image"   type="bool"   value="true"  />
    <param name="publish_truth"   type="bool"   value="false" />
    <param name="publish_est"     type="bool"   value="true" />
    <param name="camera"          type="string" value="ueye_enac_ceiling_1"/>
    <param name="detect_rgb"      type="bool"   value="false"  />
    <param name="detect_min_area" type="int"    value="24"    />

    <remap from="/camera/image_raw" to="/ueye_enac_ceiling_1/image_raw" />
  </node>

  <node pkg="tf" type="static_transform_publisher" name="world_to_cam_link_publisher"
	args="1 1 2.9 0 1.57079632679 0 world camera_link 100"/>
  <node pkg="tf" type="static_transform_publisher" name="cam_link_to_cam_optical_frame"
	args="0 0 0 -0.5 0.5 -0.5 0.5 camera_link camera_optical_frame 100"/>
  <node pkg="tf" type="static_transform_publisher" name="world_to_map_publisher"
	args="0 0 0 0 0 0 world map 100"/>
  
</launch>

